### PR TITLE
Enable lazy activation of stats output

### PR DESCRIPTION
**Pull request by simonihmig**
Currently CONCAT_STATS needs to be set at instantiation time. This enables to set it afterwards, just before the build.

### CODE REVIEW

File: lib/stats-output.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  constructor(inputNode, options) {
+    return super(inputNode, {
+      label: options.name,
+      baseDir: options.dir,
+      force: false
+    });
+  }
+
+  build() {
+    this._shouldSync = this.isEnabled();
```
Comment (simonihmig) [inline lib/stats-output.js]: Downside here is that this is using an apparently private property to activate the debug output (used to write a copy of the files). But I didn't find a better way, other than fiddling around with `BROCCOLI_DEBUG`...

Comment (stefanpenner) [reply lib/stats-output.js]: ya, but I think its fine.

**Comment by stefanpenner** [comment]
released as v3.7.0 🎉

File: lib/stats-output.js (hunk 1 above)
Comment (rwjblue) [reply lib/stats-output.js]: I agree that it’s fine, but I’d love a test case added to broccoli-debug’s test suite to make sure I don’t break the basic concept accidentally.

**Pull request merged**